Entities/Actor.cpp

```diff
@@ -617,9 +617,16 @@ void Actor::SetTeam(int team)
 
 void Actor::SetControllerMode(Controller::InputMode newMode, int newPlayer)
 {
+
+	Controller::InputMode previousMode = m_Controller.GetInputMode();
+	int previousPlayer = m_Controller.GetPlayer();
+
     m_Controller.SetInputMode(newMode);
     m_Controller.SetPlayer(newPlayer);
 
+	RunScriptedFunctionInAppropriateScripts("OnSelect", false, false, {}, {std::to_string(previousMode), std::to_string(previousPlayer) });
```

Comment:
I don't think OnSelect is a good name for this, since it's also called when deselecting someone. Try to make the name more clear, I guess, on when this is triggering.

Reply:
Aye, I thought about this one after submitting it.
Given this will also be triggered a fair bit during stuff like craft load population and all.
How does `OnSwitch` or `OnControllerSwitch` sound?

Reply:
I'm in favour of being descriptive, cause it means modders won't have to spend time digging for information on what the heck this does.
So yeah, OnControllerSwitch or OnControlModeChange or something like that is better

Reply:
Resolved by 5d41268308927e55d9f05b24b200994ddc6af2e0.